vision/deskew/within_bounds.py

```diff
@@ -0,0 +1,243 @@
+        (point2[0] - point1[0]) * (point3[1] - point2[1])
+    )
+
+    if val > 0:
+        to_return = 1  # Clockwise
+    elif val < 0:
+        to_return = 2  # Counterclockwise
+    # If val == 0, then to_return = 0
+
+    return to_return
+
+
+def lines_intersect(line1: list[list[float]], line2: list[list[float]]) -> bool:
+    """
+    Returns True if line1 and line2 intersect and False otherwise.
+
+    Parameters
+    ----------
+    line1: list[list[float]]
+        The first line that may or may not intersect with line2.
+        A 2D array that contains two points and each point is made up of
+        x and y coordinates.
+    line2: list[list[float]]
+        The second line that may or may not intersect with line1.
+        A 2D array that contains two points and each point is made up of
+        x and y coordinates.
+
+    Returns
+    -------
+    to_return: bool
+        Returns True if line1 and line2 intersect and False otherwise.
+
+    References
+    ----------
+    https://www.geeksforgeeks.org/check-if-two-given-line-segments-intersect/
+    """
+
+    to_return: bool = False
+
+    # Find the 4 orientations required for the general and special cases
+    o1: int = find_orientation(line1[0], line1[1], line2[0])
+    o2: int = find_orientation(line1[0], line1[1], line2[1])
+    o3: int = find_orientation(line2[0], line2[1], line1[0])
+    o4: int = find_orientation(line2[0], line2[1], line1[1])
+
+    if (o1 != o2) and (o3 != o4):  # General Case
+        to_return = True
+    else:  # Special Cases
+        if (o1 == 0) and on_line(line1[0], line2[0], line1[1]):
+            to_return = True
+        elif (o2 == 0) and on_line(line1[0], line2[1], line1[1]):
+            to_return = True
+        elif (o3 == 0) and on_line(line2[0], line1[0], line2[1]):
+            to_return = True
+        elif (o4 == 0) and on_line(line2[0], line1[1], line2[0]):
+            to_return = True
+
+    return to_return
+
+
+def inside_bounds(boundary: list[list[float]], location_point: list[float]) -> bool:
+    """
+    Returns True if the location_point is within the bounds of the shape
+    created by boundary. Returns False otherwise.
+
+    Parameters
+    ----------
+    boundary: list[list[float]]
+        Contains a number of points which create a shape.
+        A 2D array which contains a number of points where each point
+        is made up of x and y coordinates.
+    location_point: list[float]
+        A point that may or may not be located within the shape created
+        by boundary.
+        An Array that contains the x and y coordinates of the point.
+
+    Returns
+    -------
+    is_in_bounds: bool
+        Returns True if the location_point is within the bounds of the
+        shape created by boundary. Returns False otherwise.
+    """
+
+    is_in_bounds: bool = False
+
+    # Transforms boundary array to numpy array
+    new_np_boundary: NDArray[Shape["*, *"], Float32] = np.array(boundary)
```

Comment:
_Technically_ speaking, the `boundary` variable is a list and not an array. If you want slightly better names, you could use `boundary_list` for the list and `boundary_array` for the numpy array.